examples/bigquery-audit-log/bigquery_audit_log.sql

```diff
@@ -0,0 +1,187 @@
+    CAST(CEIL((TIMESTAMP_DIFF(
+      protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.endTime,
+      protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.startTime, MILLISECOND) / 1000) / 60) AS INT64)
+      AS executionMinuteBuckets,
+    CASE
+      WHEN
+        protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalProcessedBytes IS NULL
+        AND protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalSlotMs IS NULL
+        AND protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatus.error.code IS NULL
+        THEN true
+      ELSE false
+    END as cached,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalSlotMs,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalTablesProcessed,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalViewsProcessed,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalProcessedBytes,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.totalBilledBytes,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.billingTier,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobConfiguration.query,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.referencedTables,
+    protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.referencedViews
+    /* End: Extracts column specific to Query operation in BQ */
+  FROM
+    `data-analytics-pocs.billing.cloudaudit_googleapis_com_data_access_*`
+  WHERE
+    protopayload_auditlog.serviceName = 'bigquery.googleapis.com'
+    AND protopayload_auditlog.methodName = 'jobservice.jobcompleted'
+    AND protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.eventName
+    IN ( 'table_copy_job_completed',
+    'query_job_completed',
+    'extract_job_completed',
+    'load_job_completed')
+    AND DATE_DIFF( CURRENT_DATE(), DATE(protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobStatistics.startTime), month) <= 12
+)
+
+
+/* Query the audit */
+
+SELECT
+  principalEmail,
+  callerIp,
+  serviceName,
+  methodName,
+  eventName,
+  projectId,
+  jobId,
+  errorCode,
+  errorMessage,
+  STRUCT(
+    EXTRACT(MINUTE FROM startTime) AS minuteOfDay,
+    EXTRACT(HOUR FROM startTime) AS hourOfDay,
+    EXTRACT(DAYOFWEEK FROM startTime) - 1 AS dayOfWeek,
+    EXTRACT(DAYOFYEAR FROM startTime) AS dayOfYear,
+    EXTRACT(WEEK FROM startTime) AS week,
+    EXTRACT(MONTH FROM startTime) AS month,
+    EXTRACT(QUARTER FROM startTime) AS quarter,
+    EXTRACT(YEAR FROM startTime) AS year
+  ) as date,
+  createTime,
+  startTime,
+  endTime,
+  runtimeMs,
+  TRUNC(runtimeMs / 1000.0, 2) AS runtimeSecs,
```

Comment:
I think this is created in the with statement.

Reply:
I removed the calculation for runtimeSecs